### PR TITLE
Support to config operation log root dir both for kyuubi server and engine sides

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Support to config operation log root dir with kyuubi configuration both for server and engine sides.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala [hunk 1]
```diff
@@ -25,16 +25,26 @@ import java.nio.file.{Files, Path, Paths}
...
+import org.apache.kyuubi.config.KyuubiConf.{ENGINE_OPERATION_LOG_DIR_ROOT, SERVER_OPERATION_LOG_DIR_ROOT}
 import org.apache.kyuubi.operation.OperationHandle
 import org.apache.kyuubi.session.SessionHandle
 
 object OperationLog extends Logging {
+  val conf = KyuubiConf()
 
-  def LOG_ROOT: String = if (Utils.isTesting) {
-    "target/operation_logs"
+  def SERVER_LOG_ROOT: String = if (Utils.isTesting) {
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala:36]: move this to session managers.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala [hunk 2]
```diff
@@ -31,6 +32,12 @@ class KyuubiSessionManager private (name: String) extends SessionManager(name) {
 
   def this() = this(classOf[KyuubiSessionManager].getSimpleName)
 
+  override def LOG_ROOT: String = if (Utils.isTesting) {
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala:35]: we don't need this anymore, we can set SERVER_OPERATION_LOG_DIR_ROOT in test or maven pom file

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala [hunk 3]
```diff
@@ -31,6 +32,12 @@ class KyuubiSessionManager private (name: String) extends SessionManager(name) {
...
+    "target/server_operation_logs"
+  } else {
+    conf.get(SERVER_OPERATION_LOG_DIR_ROOT).getOrElse("server_operation_logs")
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala:38]: it seems that we need to pre-touch or validate the root dir

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala (hunk 2 above)
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala:35]: operationLogRoot

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 4]
```diff
@@ -41,6 +41,8 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
 
   @volatile private var shutdown = false
 
+  def LOG_ROOT: String
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:44]: // initialize it in `def initialize(conf: KyuubiConf)`
protected var _operationLogRoot 
def operationLogRoot = _operationLogRoot

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 5]
```diff
@@ -41,6 +43,20 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
 
   @volatile private var shutdown = false
 
+  protected var _operationLogRoot: String = null
```
Comment (turboFei) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:46]: have to set it to null, otherwise, we need override it in sub class

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 6]
```diff
@@ -41,6 +43,20 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
...
+
+  def operationLogRoot: String = _operationLogRoot
+
+  private def initOperationLogRootDir(): Unit = {
```
Comment (turboFei) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:50]: We need init it in session manager, but I am not sure whether throw the exception if it failes

Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:50]: maybe we can make operationLogRoot and Option[String], if we fail to initialize the _operationLogRoot, then we don't initialize the OperationLog for operations and just return the result

Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:50]: or maybe we just remove this step for now

Comment (zhang1002) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:50]: i think it should throw the exception or logger it in the log.

when user set this configuration item by themself, this dir perhaps has no permission for KYUUBI to create. It should let user to know.

Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:50]: > i think it should throw the exception or logger it in the log.
> 
> when user set this configuration item by themself, this dir perhaps has no permission for KYUUBI to create. It should let user to know.

It seems that you misunderstand something here, the config is from sessionManager not from users

Comment (zhang1002) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:50]: user set this configuration item.
KYUUBI create this dir.

**Review by yaooqinn — APPROVED** [review]

**Comment by turboFei** [comment]
thanks for your review, merged to master
